src/NUnitEngine/nunit.engine.tests/Internal/SettingsStoreTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.IO;
+using System.Xml.Schema;
+using NUnit.Framework;
+
+namespace NUnit.Engine.Internal.Tests
+{
+    [TestFixture]
+    public class SettingsStoreTests
+    {
+        private string settingsFile;
+        private SettingsStore settings;
+
+        [SetUp]
+        public void SetUp()
+        {
+            settingsFile = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
```

Comment:
That's not obsessive, it's basic. However, I would look into using a mock rather than creating a file - at least to evaluate whether that is a practical approach. I can live with the file, however, so long as it's deleted.